src/impl/tpls/KokkosBlas_gesv_tpl_spec_decl.hpp

```diff
@@ -44,6 +44,216 @@
 #ifndef KOKKOSBLAS_GESV_TPL_SPEC_DECL_HPP_
 #define KOKKOSBLAS_GESV_TPL_SPEC_DECL_HPP_
 
+namespace KokkosBlas {
+namespace Impl {
+  template<class AViewType, class BViewType>
+  inline void gesv_print_specialization() {
+      #ifdef KOKKOSKERNELS_ENABLE_CHECK_SPECIALIZATION
+        #ifdef KOKKOSKERNELS_ENABLE_TPL_MAGMA
+          printf("KokkosBlas::gesv<> TPL MAGMA specialization for < %s , %s >\n",typeid(AViewType).name(),typeid(BViewType).name());
+        #else
+          #ifdef KOKKOSKERNELS_ENABLE_TPL_BLAS
+            printf("KokkosBlas::gesv<> TPL Blas specialization for < %s , %s >\n",typeid(AViewType).name(),typeid(BViewType).name());
+          #endif        
+        #endif
+      #endif
+  }
+}
+}
+
+// Generic Host side BLAS (could be MKL or whatever)
+#ifdef KOKKOSKERNELS_ENABLE_TPL_BLAS
+#include<KokkosBlas_Host_tpl.hpp>
+
+namespace KokkosBlas {
+namespace Impl {
+
+#define KOKKOSBLAS_DGESV_BLAS( LAYOUT, MEM_SPACE, ETI_SPEC_AVAIL ) \
+template<class ExecSpace> \
+struct GESV< \
+     Kokkos::View<double**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+                  Kokkos::MemoryTraits<Kokkos::Unmanaged> >, \
+     Kokkos::View<double**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+                  Kokkos::MemoryTraits<Kokkos::Unmanaged> >, \
+     true, ETI_SPEC_AVAIL> { \
+  typedef double SCALAR; \
+  typedef Kokkos::View<SCALAR**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+      Kokkos::MemoryTraits<Kokkos::Unmanaged> > AViewType; \
+  typedef Kokkos::View<SCALAR**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+      Kokkos::MemoryTraits<Kokkos::Unmanaged> > BViewType; \
+ \
+  static void \
+  gesv (const char pivot[], \
+        const AViewType& A, \
+        const BViewType& B) { \
+    \
```

Comment:
You should get all necessary arrays as arguments instead of allocating inside of the function. When this code is often used, such internal allocation will be harmful.

Reply:
@kyungjoo-kim Thanks for suggestion. You also mean the user should be responsible for allocating the pivoting array, right? Then, we should have two interfaces one for gesv with pivoting and one for gesv without pivoting.

Reply:
no. if the provided pivot view is a trivial object (zero extent), then you can consider it as no pivot version.

Reply:
^^^ what Kyungjoo said -- the BLAS doesn't allocate, so users will be surprised if this function allocates.

Reply:
I also recommend making this change @kyungjoo-kim suggested.